### ISSUE TITLE
xrdfs stat succeed only after certain time

### ISSUE THREAD

**Issue by TomasJavurek** [xrootd/xrootd#912]
Hello, 

when we do physical check after uploading a file using xrdfs stat, it gives use different results with respect to other protocols for first few seconds. The result of other protocols is turstable, so the xrootd claims False, although the file is physically present at the storage. This behaviour was observed at DPM architecture for now. But I suspect it happens also on other architectures.

Kind regards, Rucio team.

2019-02-25 14:36:34,001    INFO    Successfully uploaded file test.uploadapi.3278def9-07aa-4bcc-a047-493677425533
root://fal-pygrid-30.lancs.ac.uk:1094//dpm/lancs.ac.uk/home/atlas/atlasscratchdisk/rucio/user/twegner/b6/96/test.uploadapi.3278def9-07aa-4bcc-a047-493677425533
2019-02-25 14:36:34,100    INFO    Exists: False
srm://fal-pygrid-30.lancs.ac.uk:8446/srm/managerv2?SFN=/dpm/lancs.ac.uk/home/atlas/atlasscratchdisk/rucio/user/twegner/b6/96/test.uploadapi.3278def9-07aa-4bcc-a047-493677425533
2019-02-25 14:36:34,453    INFO    Exists: True
root://fal-pygrid-30.lancs.ac.uk:1094//dpm/lancs.ac.uk/home/atlas/atlasscratchdisk/rucio/user/twegner/b6/96/test.uploadapi.3278def9-07aa-4bcc-a047-493677425533
2019-02-25 14:36:35,490    INFO    Exists: False
srm://fal-pygrid-30.lancs.ac.uk:8446/srm/managerv2?SFN=/dpm/lancs.ac.uk/home/atlas/atlasscratchdisk/rucio/user/twegner/b6/96/test.uploadapi.3278def9-07aa-4bcc-a047-493677425533
2019-02-25 14:36:35,846    INFO    Exists: True
....
root://fal-pygrid-30.lancs.ac.uk:1094//dpm/lancs.ac.uk/home/atlas/atlasscratchdisk/rucio/user/twegner/b6/96/test.uploadapi.3278def9-07aa-4bcc-a047-493677425533
2019-02-25 14:36:43,834    INFO    Exists: True

**Comment by abh3** [comment]
I suspect that the other protocols are synchrnous and wait until the DPM 
database is updated. XRootD is asynchrnous so issuing a stat right after 
an upload likely does not give enough time for the DPM database to be 
updated. I will check with Fabrizio on how to address this.

**Comment by TomasJavurek** [comment]
The physical checks with the different protocols were issued always at almost same time. e.g.: test at 14:36:34 is giving True for srm protocl, while root is False.

**Comment by abh3** [comment]
I don't see how that contradicts my hypothesis.

**Comment by wyang007** [comment]
for the record: Tomas and Wei testing the same procedure against a SLAC Xrootd installation and didn't see this problem.

**Comment by TomasJavurek** [comment]
Hi Wei, yes, but this is because of work around that we implemented in Rucio. This problem is hidden by 6 attempts that Rucio does always after two times longer interval wrt previous attempt.

**Comment by ffurano** [comment]
I see SRM here in the mix, which meant using the very old DPM components. SRM is not supported since many years and the components in that mix were prone to this kind of things.
Recent versions of DPM are not known to have this kind of issues.

We would like to understand if this issue is still valid, also given the fact that the DPM support has lowered very much in the last years.
Can you still see this issue? Which version of DPM are you running?

**Comment by amadio** [comment]
@TomasJavurek Could you please try again with the latest version of XRootD and report here if this is still an issue? I wonder if this issue can now be considered obsolete and closed.

**Comment by amadio** [comment]
Closing due to no activity. Please feel free to open a new ticket in case this is still a problem.
